### PR TITLE
Yos data

### PR DESCRIPTION
**Pull request by yostienaNegsh2336**
task 1 Log on request , log on response 
Task 2 connection request and connection response

### CODE REVIEW

**Comment by yostienaNegsh2336** [comment]
done

File: Tower.cpp [hunk 1]
```diff
@@ -58,5 +92,34 @@ int main(){
...
+    std::cin>>input;
+    ConnectionRequest connectionRequest = {
+        99, "HIAB", "ET-AUE", "A0B1C2", "HAAB", "HASC"
+    };
+
+    switch (input)
+    {
+        case 1:
+            std::cout<<"\n\nSending Connection Request to Aircraft ...\n\n";
+            tower.send(connectionRequest.toString());
```
Comment (takenaldo) [inline Tower.cpp:113]:
```suggestion
           connectionRequest.message = "ACCEPTED";
            tower.send(connectionRequest.toString());
```

File: Tower.cpp [hunk 2]
```diff
@@ -58,5 +92,34 @@ int main(){
...
+             if (connectionRequest.responseRequired)
+            {
+                tower.start_receiving();
+            }
+            
+            break;
+        
+        case 2:
+            break;
```
Comment (takenaldo) [inline Tower.cpp:126]:
```suggestion
           connectionRequest.message = "REJECTED";
           tower.send(connectionRequest.toString());
            break;
```

File: Aircraft.cpp [hunk 3]
```diff
@@ -55,6 +57,74 @@ void Aircraft::send(DataLinkMessage message){
...
+        AircraftReceiver receiver{receiver_ip, ctx, zmq::socket_type::pull};
+        std::string received_message = receiver.recieve();
+        std::cout<<"Connected ======== "<<received_message<<"\n\n\n";
+
+        DataLinkMessage datalinkMessage{received_message};
+         std::cout << "Received Message ID: " << datalinkMessage.id << std::endl;
+        std::cout << "Expected DM_CONNECTION_RESPONSE ID: " << DataLinkMessage::DM_CONNECTION_RESPONSE << std::endl;
+
+        if (datalinkMessage.id == DataLinkMessage::DM_CONNECTION_RESPONSE){
+            std::cout << "\n\nCONNECTION REQUEST RECEIVED\n";
```
Comment (takenaldo) [inline Aircraft.cpp:110]:
```suggestion
            std::cout << "\n\nCONNECTION REQUEST RECEIVED\n";
            std::cout<<"Connection Status : "<<datalinkMessage.message;
```

File: Aircraft.cpp [hunk 4]
```diff
@@ -55,6 +57,74 @@ void Aircraft::send(DataLinkMessage message){
...
+        }
+        // catch (const std::invalid_argument& e) {
+        //     std::cerr << "Invalid Argument: " << e.what() << std::endl;
+        // }
+    }
+
+    //// //  =========================connection Response 
+
+
+    void Aircraft::start_Response(){
```
Comment (takenaldo) [inline Aircraft.cpp]: use CamelCase
```suggestion
    void Aircraft::startResponse(){
```

File: Aircraft.cpp [hunk 5]
```diff
@@ -55,6 +57,74 @@ void Aircraft::send(DataLinkMessage message){
     send(message.toString());
 }
 
+
+// Start Receiving Messages
+void Aircraft::start_Receiving() {
```
Comment (takenaldo) [inline Aircraft.cpp]:
```suggestion
void Aircraft::startReceiving() {
```

**Pull request closed without merging**